### PR TITLE
Move admin routers from /admin to /-/admin

### LINKED ISSUE #32181: Allow user name "admin" to be created
### Description

auth.go:570:createUserInContext() [E] CreateUser: name is reserved [name: admin]

Why can't we create a user named admin? This causes issues with my SSO setup where I use "admin" across all services, so now Gitea breaks when I login with SSO it tries to create that user and fails.

### Gitea Version

1.19.0 and 1.22.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Myself in docker

### Database

MySQL/MariaDB

### PR DESCRIPTION
**Pull request by lunny**
Resolve #32181

### CODE REVIEW

**Comment by lafriks** [comment]
Maybe move it to `/-/admin/` instead so in the future some others could be moved/added there as well?

**Comment by lafriks** [comment]
Test failures are related

**Review by sundaram2021 — COMMENTED** [review]
## review
good to merge
I like the way you did it :)

**Review by delvh — APPROVED** [review]
Two things:
1. I hope we didn't overlook any URLs still pointing to the old path. Oh well, there will be bug reports if that happens.
2. I think this will be one of the largest UI-link breaking actions we have ever done. Let's hope our users muscle memory can be overwritten soon enough (for those that typed the URL themselves instead of finding a link)

File: routers/web/admin/applications.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ var (
 func newOAuth2CommonHandlers() *user_setting.OAuth2CommonHandlers {
 	return &user_setting.OAuth2CommonHandlers{
 		OwnerID:            0,
-		BasePathList:       fmt.Sprintf("%s/admin/applications", setting.AppSubURL),
-		BasePathEditPrefix: fmt.Sprintf("%s/admin/applications/oauth2", setting.AppSubURL),
+		BasePathList:       fmt.Sprintf("%s%s/applications", setting.AppSubURL, setting.AdminRouterPrefix),
+		BasePathEditPrefix: fmt.Sprintf("%s%s/applications/oauth2", setting.AppSubURL, setting.AdminRouterPrefix),
```
Comment (wxiaoguang) [inline routers/web/admin/applications.go]: The const bring no benefit but only make the code more complex.

Even if you introduce a const, there are still `/-/admin` strings in templates.

----

* With a const, it needs to search both `AdminRouterPrefix` and `/-/admin`.
* Without a const, it only needs to search  `/-/admin` which is simpler.

Comment (lunny) [reply routers/web/admin/applications.go]: ea92b94a0e6873e24747360c376d25e2d606069e

**Comment by yp05327** [comment]
Seems like a breaking change?

**Comment by lunny** [comment]
> Seems like a break change?

If the user always uses the menu to enter the admin page, it should not be considered as a break change.

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Conflicts caused by my PR have been resolved.

**Pull request merged**